src/Service/ComponentenCatalogusService.php

```diff
@@ -456,54 +499,72 @@ public function handleGithubComponentRepo(Source $source, array $componentArray)
 
         return $repository;
 
-    }//end handleGithubComponentRepo()
+    }//end handleComponentRepo()
 
 
     /**
      * Get all components of the given source.
      *
-     * @param Source $source         The given source
+     * @param Source $defaultSource  The (default source) componenten catalogus.
      * @param array  $componentArray The component array
      *
      * @return ObjectEntity|null
      * @throws \Exception
      */
-    public function handleComponent(Source $source, array $componentArray): ?ObjectEntity
+    public function handleComponent(Source $defaultSource, array $componentArray): ?array
     {
-
         $parsedUrl = \Safe\parse_url($componentArray['repositoryUrl']);
         if (key_exists('host', $parsedUrl) === false) {
             return null;
         }
 
+        // Get the domain of the repository url.
         $domain = $parsedUrl['host'];
+
         switch ($domain) {
+            // If the domain is from github get the github source.
         case 'github.com':
-            return $this->handleGithubComponentRepo($source, $componentArray);
-                break;
+            $source = $this->resourceService->getSource($this->configuration['githubSource'], 'open-catalogi/open-catalogi-bundle');
+            break;
+            // If the domain is from gitlab get the gitlab source.
         case 'gitlab.com':
+            $source = $this->resourceService->getSource($this->configuration['gitlabSource'], 'open-catalogi/open-catalogi-bundle');
             break;
         default:
+            $source = null;
+            $this->pluginLogger->info('We don\'t support this repository with domain: '.$domain);
             break;
-        }//end switch
+        }
 
-        return null;
+        if ($source instanceof Source === false) {
+            return null;
```

Comment:
Maybe we want some error logs when we return on places like this to debug if something goes wrong?

Reply:
The resourceService->getSource function will create an error log when no source is found.
But yess another log here wouldn't be bad